hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/test/java/org/apache/hadoop/yarn/server/router/webapp/TestFederationInterceptorRESTRetry.java

```diff
@@ -380,10 +385,20 @@ public void testGetNodeOneBadOneGood()
   @Test
   public void testGetNodesOneBadSC() throws Exception {
 
+    // allowPartialResult = true,
+    // We tolerate exceptions and return normal results
+    interceptor.setAllowPartialResult(true);
     setupCluster(Arrays.asList(bad2));
 
-    LambdaTestUtils.intercept(YarnRuntimeException.class, "RM is stopped",
```

Comment:
Can we keep the old test and add a new one with the partial enabled?

Reply:
Thank you very much for reviewing the code, I will add new unit tests.